py-polars/polars/internals/dataframe/frame.py

```diff
@@ -4252,7 +4256,12 @@ def with_column(self, column: pli.Series | pli.Expr) -> DataFrame:
         └──────┴─────┘
 
         """
-        return self.lazy().with_column(column).collect(no_optimization=True)
+        warnings.warn(
+            "`with_column` has been deprecated in favor of `with_columns`."
+            " This method will be removed in the future",
+            category=DeprecationWarning,
+        )
```

Comment:
Add `stacklevel=2` so that people can see which line of their code raises the warning.

```suggestion
            category=DeprecationWarning,
            stacklevel=2,
        )
```